File: backend/internal/domain/booking.go
package domain

import "context"

type Booking struct {
	ID                 int64   `db:"id" json:"id"`
	ClientID           int64   `db:"client_id" json:"clientId"`
	SlotID             int64   `db:"slot_id" json:"-"`
	Status             string  `db:"status" json:"status"`
	RentalSelected     bool    `db:"rental_selected" json:"rentalSelected"`
	CreatedAt          string  `db:"created_at" json:"createdAt"`
	CancellationReason *string `db:"cancellation_reason" json:"cancellationReason,omitempty"`
	Slot               *Slot   `json:"slot,omitempty"`
}

type BookingRepository interface {
	InsertBooking(ctx context.Context, clientID, slotID int64, rentalSelected bool) (Booking, error)
	QueryBookingsByClient(ctx context.Context, clientID int64, statusFilter string) ([]Booking, error)
	GetBookingByID(ctx context.Context, id int64) (Booking, error)
	UpdateBookingStatus(ctx context.Context, id int64, status string, reason *string) error
	CancelBookingTx(ctx context.Context, bookingID, slotID int64) error
}

File: mobile/lib/core/models/booking.dart
import 'slot.dart';

class Booking {
  final int id;
  final int clientId;
  final Slot slot;
  final String status;
  final bool rentalSelected;
  final String createdAt;
  final String? cancellationReason;

  const Booking({
    required this.id,
    required this.clientId,
    required this.slot,
    required this.status,
    required this.rentalSelected,
    required this.createdAt,
    this.cancellationReason,
  });

  bool get isActive => status == 'активна';
  bool get isCancelledByClient => status == 'отменена клиентом';
  bool get isCancelledByWorkshop => status == 'отменена мастерской';

  factory Booking.fromJson(Map<String, dynamic> json) => Booking(
    id: json['id'] as int,
    clientId: json['clientId'] as int,
    slot: Slot.fromJson(json['slot'] as Map<String, dynamic>),
    status: json['status'] as String,
    rentalSelected: json['rentalSelected'] as bool,
    createdAt: json['createdAt'] as String,
    cancellationReason: json['cancellationReason'] as String?,
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'clientId': clientId,
    'slot': slot.toJson(),
    'status': status,
    'rentalSelected': rentalSelected,
    'createdAt': createdAt,
    'cancellationReason': cancellationReason,
  };
}
